src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -1110,7 +1113,7 @@ private void RemoveAdalUser(IAccount account)
                 LegacyCachePersistence,
                 ClientId,
                 account.Username,
-                account.HomeAccountId.Identifier);
+                account.HomeAccountId?.Identifier);
         }
```

Comment:
when would there not be a homeAccountId?

Reply:
TokenCache line 953 (users without clientinfo):  

accounts.Add(new Account(null, user.DisplayableId, envFromRequest));